test/jdk/java/util/zip/ZipFile/CenSizeTooLarge.java

```diff
@@ -49,11 +50,11 @@ public class CenSizeTooLarge {
     static final int MAX_CEN_SIZE = Integer.MAX_VALUE - ZipFile.ENDHDR - 1;
 
     // Maximum size (unsigned short) of an extra field allowed by the standard
-    static final int MAX_EXTRA_FIELD_SIZE = 2 * Short.MAX_VALUE;
+    static final int MAX_EXTRA_FIELD_SIZE = 0XFFFF;
```

Comment:
4.4.10 file name length: (2 bytes)
   4.4.11 extra field length: (2 bytes)
   4.4.12 file comment length: (2 bytes)

       The length of the file name, extra field, and comment
       fields respectively.  The combined length of any
       directory record and these three fields SHOULD NOT
       generally exceed 65,535 bytes.

Reply:
`The combined length of any directory record and these three fields SHOULD NOT generally exceed 65,535 bytes.`

I was not aware of this 'combined length' clause. Are you suggesting we take this into account in this test? I feel that would be somewhat strange, given that this constraint seems not be be enforced at all by ZipEntry or ZipOutputStream:

```java
ZipEntry(String name):

if (name.length() > 0xFFFF) {
    throw new IllegalArgumentException("entry name too long");
}
```

```
ZipOutputStream.writeCEN:
```java
if (commentBytes != null) {
    writeBytes(commentBytes, 0, Math.min(commentBytes.length, 0xffff));
}
```

Reply:
I would clarify the comment a bit and perhaps point to the comments Martin mentions in the review above